### PR TITLE
[release-4.5] Bug 1896079: Services should not have duplicate ownerrefs

### PR DESCRIPTION
**Pull request by ecordell**
**Description of the change:**
This change adds a unit test to verify that services created for apiservices installed via OLM do not have multiple ownerrefs added, and includes https://github.com/operator-framework/operator-lifecycle-manager/commit/1476ff19c6a0c0a37a2fcdb47aa982683d37fee6 which fixes the underlying issue.

The test harness is bulky, but because this is intended to be backported, I did not attempt to refactor the related apiservice management to make testing simpler.

**Motivation for the change:**
This is to verify that the changes from this commit (https://github.com/operator-framework/operator-lifecycle-manager/commit/1476ff19c6a0c0a37a2fcdb47aa982683d37fee6) are included in 4.5.

This set of changes + the above commit will be backported to older versions, to address a report of this bug in 4.5

**Reviewer Checklist**
- [ ] Implementation matches the proposed design, or proposal is updated to match implementation
- [ ] Sufficient unit test coverage 
- [ ] Sufficient end-to-end test coverage
- [ ] Docs updated or added to `/docs` 
- [ ] Commit messages sensible and descriptive

### CODE REVIEW

File: pkg/controller/install/certresources_test.go [hunk 1]
```diff
@@ -0,0 +1,577 @@
...
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "doesn't add duplicate service ownerrefs",
```
Comment (njhale) [inline pkg/controller/install/certresources_test.go:348]: I'm having a difficult time finding where the service is actually checked for duplicate ownerreferences in the test case. @ecordell could you point it out?

**Comment by sdodson** [comment]
Putting a temporary hold on things to prioritize items that fix test flakes while we're hitting AWS quota limits. I'll clear this later today.

/hold

**Comment by ecordell** [comment]
/hold 

unit test failures look consistent

**Comment by dinhxuanvu** [comment]
@ecordell Hey Evan. Have you had the time to look into the failures?

**Comment by bparees** [comment]
pinged evan on slack, he's going to look into the failures and hopefully unstick this.

**Comment by ecordell** [comment]
/bugzilla refresh
/hold cancel

I bumped to bug to high because this can manifest as a blocked OCP update (there's an easy workaround if a cluster hits that, so I haven't bumped to urgent).

The unit test failure was due to the automatic rebasing that tide does. Manually rebasing and fixed the conflict that we saw before.

**Comment by ecordell** [comment]
/lgtm cancel

looks like there are more rebase-related issues

**Pull request merged**